### ISSUE TITLE
Huawei signin fail

### ISSUE THREAD

**Issue by FaneNego** [microg/GmsCore#1189]
Adding a account on Huawei p 40 lite is impossible

**Comment by EloYGomeZ** [comment]
If for a few days you can not connect. the case that the people of vanced have managed to login... And they should implement it here.

**Comment by qtekfun** [comment]
Same here on P40 Pro. as @EloYGomeZ says, vanced is able to connect without problems. I have reviewed some code of gsmcore but I don't find any solution.

**Comment by FaneNego** [comment]
I was told that microg vanced and microg gms core are different things, but I hope there will be a fix soon.
I also installed vanced without a problem

**Comment by EloYGomeZ** [comment]
Of course they are different things, but the call to the server for what little I could see in the code makes it the same... The same thing I am wrong and I am skipping a step.

**Comment by FaneNego** [comment]
Ok then. Keep us notified if there's a update

**Comment by EloYGomeZ** [comment]
The one who has to inform us is MicroG :) ... Let's see if they give us a solution soon.

**Comment by bubobih** [comment]
Confirming that it doesent work on all huawei devices due google sing in patch update. Also all other methods for google mobile service on huawei stop to work including lzplay.

**Comment by EloYGomeZ** [comment]
It would have to be done like Aurora Store does. Change the device for another one during the connection to the server and then return to the original once connected.

**Comment by bubobih** [comment]
thats actualy good idea but someone need to make it 🥇

**Comment by FaneNego** [comment]
Vanced team did it and works, now waiting for microg to do it

**Comment by FaneNego** [comment]
I believe this issue addresses a lot of people now that the lzplay method doesn't work for 2 weeks now

**Comment by bubobih** [comment]
still nothing???

**Comment by EloYGomeZ** [comment]
I am trying to make an adaptation for the huawei devices to be connected but I doubt they will approve it because it focuses only on connecting those devices.

**Comment by bubobih** [comment]
If u succes share your git :)

**Comment by EloYGomeZ** [comment]
If you look at my repository I am already working on it.

**Comment by bubobih** [comment]
Im very happy that someone at least trying to fix it for huawei. Im not any expert in this field but u can maybe compare it wirh yt vanced microg https://github.com/YTVanced/VancedMicroG

**Comment by FaneNego** [comment]
I am very happy that someone is trying to fix this

**Comment by Nanolx** [comment]
It's this commit in Vanced:

* https://github.com/YTVanced/VancedMicroG/commit/0ab986221fddba71027be091b47f577044b92687

You may ask @mar-v-in to have a look or come up with something different to solve the login issue for Huawei users.

Credits go to POTUS' trade war, which forced Google to ban Huawei from using Play Services.

**Comment by FaneNego** [comment]
I'm kinda new to ghithub how can I contact him or I just @mar-v-in

**Comment by EloYGomeZ** [comment]
Don't worry, in the next few days I'll publish something that I've already managed to connect. But I have to do some tests.

**Comment by FaneNego** [comment]
Thank you for your effort

**Comment by FaneNego** [comment]
Is there anything I can test?

**Comment by bubobih** [comment]
Im think not

**Comment by bubobih** [comment]
Try all https://m.youtube.com/watch?reload=9&v=4HfWs3kfCPI

**Comment by EloYGomeZ** [comment]
If that is my video and in my repository are the updates of the app for the connection with HUAWEI

**Comment by FaneNego** [comment]
I need to downgrade the sistem for the installation but I can't find how

**Comment by bubobih** [comment]
Yes its your video and your microg. A question. Does notifications work?

**Comment by bubobih** [comment]
Yes u need to downgrade lzplay version check make a problems. Im not sure can anyone edit lzplay to remove version check

**Comment by FaneNego** [comment]
How can I do that on my Huawei p 40 lite

**Comment by FaneNego** [comment]
Lzplay doesn't work

**Comment by Happyfeet01** [comment]
work it with microg and downgrade is recommend?

**Comment by FaneNego** [comment]
Anyway lzplay method can't work and the Huawei p40 lite can't be downgraded

**Comment by bubobih** [comment]
Any phone can be downgraded

**Comment by FaneNego** [comment]
I am trying to do that for the last 7 hours

**Comment by Happyfeet01** [comment]
Work the play services from MicroG with that fix? Maybe we dont need lzplay

**Comment by FaneNego** [comment]
I've tried it and the apps recognise the Google account but then a terror message pops up
"Can't signin, try again later"
(Google classroom)

**Comment by Happyfeet01** [comment]
> I've tried it and the apps recognise the Google account but then a terror message pops up
> "Can't signin, try again later"
> (Google classroom)


With downgrade and lzplay?

**Comment by FaneNego** [comment]
I tried the patched version of EloYgomeZ and the rest of the official microg apk's
In my case I can't find anywhere the firmware for my phone

**Comment by EloYGomeZ** [comment]
> I tried the patched version of EloYgomeZ and the rest of the official microg apk's
> In my case I can't find anywhere the firmware for my phone

the firmware can be found in google ... By asking in the appropriate forum of XDA.... But it is search a little usually all on the Internet.

**Comment by FaneNego** [comment]
Can you give me a link for hisuite proxi for my phone because I have been searching since yesterday
Model Huawei p40 lite 
Jny-l21

**Comment by EloYGomeZ** [comment]
I don't feel it. 

Try doing it directly with hisuite without proxy

**Comment by FaneNego** [comment]
On hisuite there's no option to downgrade

**Comment by bubobih** [comment]
@EloYGomeZ Your version work (sing in) but notifications doesent. As i understand (septembar google sing in break) are actualy huawei blacklisting, and u solve it by spoofing, but firebase still detect phone as huawei and notifications doesent arrive. This problem is on all things including aurora store where u need to enable spoof and change device model id. 

I assume that notifications use another id and detect phone as huawei blacklisted one. i hope u can maybe fix that too becuz i dont need play store or gms i need notifications for skype, steam,discord and many more. Im currently use orginal microg whitout sing in and notifications work.

**Comment by Happyfeet01** [comment]
That's correct aurora store didn't work. Also the German covid 19 App. But that is an another story.

**Comment by bubobih** [comment]
> That's correct aurora store didn't work. Also the German covid 19 App. But that is an another story.

U can bypass in Aurora, 

1. Sing in as anonimous, 
2. Go to menu find spoof
3. Choose device (i use nexus 6)
4. Singout
5. Login with real google account

**Comment by Happyfeet01** [comment]
Thanks. But I don't need this store, the Google play store works fine.

**Comment by bubobih** [comment]
Yes it work but u dont have notifications. For me notifications are 99%, play store 1% important.

**Comment by Happyfeet01** [comment]
Okay, but have you disabled the Framework?

**Comment by bubobih** [comment]
notifications doesent work anymore if u make new gms installation, its blocked by google.u can check xda mate 30 pro and xda p40 pro threads of new way to install gms and users reply complaining about notifications.

actualy i test it, with apk notification tester everything work even test over it. but in real situation using some apk, u dont get notifications.

**Comment by Happyfeet01** [comment]
Okay thanks

**Comment by jcchikikomori** [comment]
Hello. Please try my fork with patches from @YTVanced & @EloYGomeZ 
https://github.com/jcchikikomori/GmsCoreHuawei/pull/2

**Comment by chortya** [comment]
Could you please build the APKs for this fork?

**Comment by jcchikikomori** [comment]
> Could you please build the APKs for this fork?

Let me prepare a release with signed APKs provided

**Comment by chortya** [comment]
Ok, I cloned your repository and was able to build signed APK. Login is possible but the apps are not able to pick up registered Google account. Is anything else needed? Or is signature spoofing required for the account manager?

**Comment by jcchikikomori** [comment]
> Ok, I cloned your repository and was able to build signed APK. Login is possible but the apps are not able to pick up registered Google account. Is anything else needed? Or is signature spoofing required for the account manager?

I guess so. I cannot test it since i don't have any rooted devices. I'll try on a VM though

Btw here is the release:
https://github.com/jcchikikomori/GmsCoreHuawei/releases/tag/v0.2.13.203915-HUAWEI

**Comment by gghhkm** [comment]
> Of course they are different things, but the call to the server for what little I could see in the code makes it the same... The same thing I am wrong and I am skipping a step.

There not different things really, Vanced uses this microG as base and only changed a few things.

**Comment by chortya** [comment]
Any plans to implement any improvements for specifically for GMS-less Huawei phones? I would wish more robust notification support with extended logging capabilities. Or how can I debug the notification routines? It seems some of the apps are still having issues and pushing notifications only sporadically.
On the rest of the functionality (sing-in, logged in account list, usage of Google apps like Gmail, Photo etc) I assume it's not really possible without signature spoofing?

**Comment by jcchikikomori** [comment]
> Any plans to implement any improvements for specifically for GMS-less Huawei phones? I would wish more robust notification support with extended logging capabilities. Or how can I debug the notification routines? It seems some of the apps are still having issues and pushing notifications only sporadically.
> On the rest of the functionality (sing-in, logged in account list, usage of Google apps like Gmail, Photo etc) I assume it's not really possible without signature spoofing?

I've been thinking about different directions like @YTVanced did on Youtube Vanced
I think the only way is to modify Google apps and support microG

**Comment by qtekfun** [comment]
I thought the same but it's lots of work

**Comment by chris42** [comment]
> Any plans to implement any improvements for specifically for GMS-less Huawei phones? I would wish more robust notification support with extended logging capabilities. Or how can I debug the notification routines? It seems some of the apps are still having issues and pushing notifications only sporadically.

Normally you can see every incoming and outgoing message in the android log, that should help to debug. In my experience unreliable notifications are more a sign of a fallback being used, e.g. in whatsapp. With GCM it was always either working or not. If you want to jump into the code, I guess running microg on an emulator should help. Of course with each app itself you would need to contact the app developers.

BTW: would love the sing-in feature... Smashing Bohemian Rapsody to unlock your phone! ;-D

**Comment by michael1900** [comment]
any good news for us huawei consumer?

**Comment by chortya** [comment]
> any good news for us huawei consumer?

I don't think this is really a MicroG issue, probably a ban on Google server side for specific Huawei devices. So the only way is to fully spoof device signature towards Google (not only during the sign-in process), no idea if this can even be done.
See patched version from jcchikikomori but it does not really help with anything.

**Comment by gghhkm** [comment]
> > any good news for us huawei consumer?
> 
> I don't think this is really a MicroG issue, probably a ban on Google server side for specific Huawei devices. So the only way is to fully spoof device signature towards Google (not only during the sign-in process), no idea if this can even be done.
> See patched version from jcchikikomori but it does not really help with anything.

It's already been done. The YouTube Vanced team has Huawei spoofing in their build of it. https://github.com/YTVanced/VancedMicroG

**Comment by chortya** [comment]
> > > any good news for us huawei consumer?
> > 
> > 
> > I don't think this is really a MicroG issue, probably a ban on Google server side for specific Huawei devices. So the only way is to fully spoof device signature towards Google (not only during the sign-in process), no idea if this can even be done.
> > See patched version from jcchikikomori but it does not really help with anything.
> 
> It's already been done. The YouTube Vanced team has Huawei spoofing in their build of it. https://github.com/YTVanced/VancedMicroG

This is not a good example. They can do whatever they want because Vanced Youtube version is patched too. Here the idea is to use original Google apps with MicroG that spoofs Huawei signature.

**Comment by gghhkm** [comment]
> > > > any good news for us huawei consumer?
> > > 
> > > 
> > > I don't think this is really a MicroG issue, probably a ban on Google server side for specific Huawei devices. So the only way is to fully spoof device signature towards Google (not only during the sign-in process), no idea if this can even be done.
> > > See patched version from jcchikikomori but it does not really help with anything.
> > 
> > 
> > It's already been done. The YouTube Vanced team has Huawei spoofing in their build of it. https://github.com/YTVanced/VancedMicroG
> 
> This is not a good example. They can do whatever they want because Vanced Youtube version is patched too. Here the idea is to use original Google apps with MicroG that spoofs Huawei signature.
> Also Vanced has removed a lot of features in their MicroG implementation:
> Removed unnecessary features:
> Ads
> Analytics
> Car
> Droidguard
> Exposure-Notifications
> Feedback
> Firebase
> Games
> Maps
> Recovery
> Registering app permissions
> SafetyNet
> Self-Check
> Search
> TapAndPay
> Wallet
> Wear-Api

Who cares if they removed a bunch of stuff, doesn't mean you can't look at their approach to spoofing for Huawei devices and learn from it.

**Comment by chortya** [comment]
But this is exactly what jcchikikomori and EloYGomeZ did in their patched versions, they are just hardcoding device fingerprint (different btw). Vanced checks if the phone has native GMS and if not also hard codes the following fingerprint: "google/sdk_gphone_x86/generic_x86_arm:11/RPB3.200720.005/6705141:user/release-keys")

**Comment by chortya** [comment]
> > Ok, I cloned your repository and was able to build signed APK. Login is possible but the apps are not able to pick up registered Google account. Is anything else needed? Or is signature spoofing required for the account manager?
> 
> I guess so. I cannot test it since i don't have any rooted devices. I'll try on a VM though
> 
> Btw here is the release:
> https://github.com/jcchikikomori/GmsCoreHuawei/releases/tag/v0.2.13.203915-HUAWEI

any plans to build updated version? Could you please briefly explain what fingerprint are you using? Maybe it would be better to switch to the Vanced fingerprint or does it play any role?

**Comment by jcchikikomori** [comment]
> > > Ok, I cloned your repository and was able to build signed APK. Login is possible but the apps are not able to pick up registered Google account. Is anything else needed? Or is signature spoofing required for the account manager?
> > 
> > 
> > I guess so. I cannot test it since i don't have any rooted devices. I'll try on a VM though
> > Btw here is the release:
> > https://github.com/jcchikikomori/GmsCoreHuawei/releases/tag/v0.2.13.203915-HUAWEI
> 
> any plans to build updated version? Could you please briefly explain what fingerprint are you using? Maybe it would be better to switch to the Vanced fingerprint or does it play any role?

I already have. I never touched my master/main branch unless it's from microG.
You can check my pull request for the fingerprint thing
https://github.com/jcchikikomori/GmsCoreHuawei/pull/2

**Comment by jcchikikomori** [comment]
> Maybe it would be better to switch to the Vanced fingerprint or does it play any role?

No. I pulled this fingerprint from Vanced itself

**Comment by michael1900** [comment]
> > > > Ok, I cloned your repository and was able to build signed APK. Login is possible but the apps are not able to pick up registered Google account. Is anything else needed? Or is signature spoofing required for the account manager?
> > > 
> > > 
> > > I guess so. I cannot test it since i don't have any rooted devices. I'll try on a VM though
> > > Btw here is the release:
> > > https://github.com/jcchikikomori/GmsCoreHuawei/releases/tag/v0.2.13.203915-HUAWEI
> > 
> > 
> > any plans to build updated version? Could you please briefly explain what fingerprint are you using? Maybe it would be better to switch to the Vanced fingerprint or does it play any role?
> 
> I already have. I never touched my master/main branch unless it's from microG.
> You can check my pull request for all of my changes
> [jcchikikomori#2](https://github.com/jcchikikomori/GmsCoreHuawei/pull/2)

You think that this approach can be a good start to use microg with original google apps finally in future?

**Comment by jcchikikomori** [comment]
> You think that this approach can be a good start to use microg with original google apps finally in future?

Yes. Unlike @EloYGomeZ's version which is already used on Googlefier, my code is clean and manageable. In fact I can sync the changes from the parent itself (microG code)

**Comment by michael1900** [comment]
> > You think that this approach can be a good start to use microg with original google apps finally in future?
> 
> Yes. Unlike @EloYGomeZ's version which is already used on Googlefier, my version is manageable. In fact, I can sync the changes from the parent itself (microG code)
> 
> At least this fingerprint is the best to work on using microG.
> 
> Btw the signature spoofing is only the issue here, unless the device will side-load it explicitly (rooted)


With your version push notifications work with all apps like with standard microg?

**Comment by jcchikikomori** [comment]
> > > You think that this approach can be a good start to use microg with original google apps finally in future?
> > 
> > 
> > Yes. Unlike @EloYGomeZ's version which is already used on Googlefier, my version is manageable. In fact, I can sync the changes from the parent itself (microG code)
> > At least this fingerprint is the best to work on using microG.
> > Btw the signature spoofing is only the issue here, unless the device will side-load it explicitly (rooted)
> 
> With your version push notifications work with all apps like with standard microg?

Yes @michael1900 
I had some apps working, except Skype (Lite version works though)

**Comment by ale5000-git** [comment]
Everything should already be fixed on the latest version (microG can now spoof another device) so I close it for now.
Open a new issue if needed.

**Comment by jcchikikomori** [comment]
> Everything should already be fixed on the latest version (microG can now spoof another device) so I close it for now. Open a new issue if needed.

Indeed. In my fork, i'll revert my own device spoof implementation
